From 4613accc2e22395bb77bdf612fcdf90bf65f230f Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Fri, 23 May 2025 15:38:56 -0700
Subject: [PATCH] build: handle depending on development releases of GLib

This allows us to require an odd version number indicating unstable release
for the necessary API additions.
---
 meson.build | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 7c996ea..86f4495 100644
--- a/meson.build
+++ b/meson.build
@@ -213,8 +213,11 @@ if lua_found
 endif
 
 glib_version_arr = glib_req_version.split('.')
-glib_major_version = glib_version_arr[0]
-glib_minor_version = glib_version_arr[1]
+glib_major_version = glib_version_arr[0].to_int()
+glib_minor_version = glib_version_arr[1].to_int()
+if glib_minor_version % 2 == 1
+  glib_minor_version = glib_minor_version + 1
+endif
 
 project_c_args = [
   '-DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_@0@_@1@'.format(glib_major_version, glib_minor_version),
-- 
GitLab

